src/sources/file/mod.rs

```diff
@@ -327,6 +327,11 @@ fn create_event(
         event.as_mut_log().insert(host_key, hostname.clone());
     }
 
+    // Add source type
+    event
+        .as_mut_log()
+        .try_insert(event::log_schema().source_type_key(), "file");
+
```

Comment:
Would it make sense to add methods beside `Event::from` that add the source type right there? That would eliminate these repeated calls to insert the source type later.

Reply:
It would require more modifications/methods for a benefit of removing a function call at best. Although we could insert, instead of try_insert, immediately after that since we know what fields are present.

Reply:
Hmmm, ok, I had forgotten that the keys could/will be per-source. That makes it more complicated.